src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java

```diff
@@ -155,15 +163,30 @@ public void onFailure(Exception e) {
             }
         }
 
-        if (Stream.of(name, type, llm, tools, parameters, memory, appType).allMatch(x -> x != null)) {
+        // Case when modelId is present in previous node inputs
+        if (llmModelId == null) {
+            llmModelId = getLlmModelId(previousNodeInputs, outputs);
+        }
+
+        // Case when modelId is not present at all
+        if (llmModelId == null) {
+            registerAgentModelFuture.completeExceptionally(
```

Comment:
Can we handle the second null check on line 174 and completeExceptionally/return there?

Reply:
We can't. The reason being the chronological order. If we didn't find the `llmModelId` in the user inputs, we will go and find it in the previousNodeInputs. If it's not present in both then we will `completeExceptionally`.